src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClientBase.h

```diff
@@ -65,6 +67,33 @@ namespace client
     /* Non-template base client class that contains main Aws Client Request pipeline logic */
     class SMITHY_API AwsSmithyClientBase
     {
+    private:
+        const char* ALLOC_TAG{"AwsSmithyClientBase"};
+
+
+        void deepCopyClientConfiguration(const AwsSmithyClientBase& target)
+        {
+            //first create copy from target
+            m_clientConfig = Aws::MakeShared<Aws::Client::ClientConfiguration>(target.m_serviceName.c_str(), *target.m_clientConfig);
```

Comment:
i dont think this is going to work the way we want it to because of slicing here we declare a pointer of `ClientConfiguration` not `ServiceClientConfigurationT`. in `AwsSmithyClientT` we call it as such

```
m_clientConfiguration{*static_cast<ServiceClientConfigurationT*>(AwsSmithyClientBase::m_clientConfig.get())},
```

since this is _NOT_ a `ServiceClientConfigurationT` we enter UB territory.

trying to think of a creative solution around this let me talk to you offline about this